Python/pywarpx/_libwarpx.py

```diff
@@ -464,9 +181,15 @@ def finalize(self, finalize_mpi=1):
         Call finalize for WarpX and AMReX. Registered to run at program exit.
 
         '''
+        # TODO: simplify, part of pyAMReX already
         if self.initialized:
-            self.libwarpx_so.warpx_finalize()
-            self.libwarpx_so.amrex_finalize(finalize_mpi)
+            del self.warpx
+            # The call to warpx_finalize causes a crash - don't know why
+            #self.libwarpx_so.warpx_finalize()
```

Comment:
This segfaults. Let me attempt to remove some logic from our constructor - I bet it operates on an already removed instance.

Generally, in Python, we want to have no heavy logic in destructors and thus I move it to `Finalize()`.

Reply:
Ok, I tried reworking this and this will need more work later on. Too many cross-dependencies between living Python objects and the instance destruction order for now.